File: Editor/src/main/java/org/turbomedia/turboedit/editor/windows/preferences/AddRenderServerWindow.java
package org.turbomedia.turboedit.editor.windows.preferences;

import javafx.scene.Scene;
import javafx.stage.Modality;
import javafx.stage.Stage;
import org.turbomedia.turboedit.editor.panes.preferences.EditRenderServerPane;
import org.turbomedia.turboedit.editor.renderer.FileResolveMethod;
import org.turbomedia.turboedit.editor.renderer.RenderServerEntry;

import static org.turbomedia.turboedit.editor.Editor.ICON;
import static org.turbomedia.turboedit.editor.Editor.TITLE;
import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public class AddRenderServerWindow extends Stage {
    public AddRenderServerWindow(String responseID) {
        setTitle(TITLE + " - " + GetText("title.add_render_server"));
        getIcons().add(ICON);
        setWidth(500);

        setScene(new Scene(new EditRenderServerPane(this, new RenderServerEntry("", "", FileResolveMethod.STREAMING, false, false), responseID)));

        setResizable(false);
        centerOnScreen();
        initModality(Modality.APPLICATION_MODAL);
        showAndWait();
    }
}

File: Editor/src/main/kotlin/turboedit/editor/windows/preferences/AddRenderServerWindow.kt
package turboedit.editor.windows.preferences

import javafx.scene.Scene
import javafx.stage.Modality
import javafx.stage.Stage
import turboedit.editor.Editor
import turboedit.editor.misc.Locale
import turboedit.editor.panes.preferences.EditRenderServerPane
import turboedit.editor.renderer.FileResolveMethod
import turboedit.editor.renderer.RenderServerEntry

class AddRenderServerWindow(responseID: String?) : Stage() {
    init {
        title = Editor.Companion.TITLE + " - " + Locale.getText("title.add_render_server")
        icons.add(Editor.Companion.ICON)
        isResizable = false
        scene = Scene(
            EditRenderServerPane(
                this, RenderServerEntry("", "", "", FileResolveMethod.STREAMING, false, false), responseID
            )
        );
        width = 500.0

        centerOnScreen()
        initModality(Modality.APPLICATION_MODAL)
        showAndWait()
    }
}
